src/enums.h

```diff
@@ -51,48 +72,87 @@ std::array<std::string_view, size_t(EnumType::FINAL)> enum_strarray(
- * the names in the ENUMTYPE enum class
+ * a std::array<std::string_view, TYPE(ENUMTYPE::FINAL)> with all of
+ * the names in the ENUMTYPE enum class (bar FINAL)
+ *
+ * Additionally, will fill a local namespace "enumtyps" with
+ * a std::array<ENUMTYPE, TYPE(ENUMTYPE::FINAL)> with all of
+ * the types in the ENUMTYPE enum class (bar FINAL)
  *
- * Additionally, will generate a inlined "toString" function
+ * Additionally, will generate a constexpr "toString" function
  * that takes a ENUMTYPE object and returns either its partial
- * name or "BAD ENUMTYPE" as a string
+ * name or "BAD ENUMTYPE" as a string_view
  *
- * Additionally, will generate a inlined "toENUMTYPE" function
- * that takes a std::string and returns a corresponding ENUMTYPE
+ * Additionally, will generate a constexpr "toENUMTYPE" function
+ * that takes a std::string_view and returns a corresponding ENUMTYPE
  * object or ENUMTYPE::FINAL if the object is bad
+ * 
+ * Additionally, will generate intuitive std::istream& and
+ * std::ostream& operators
  *
  * Use the "good_enum(ENUMTYPE)" template function to check
- * if the enum classs object is any good
- *
- * Will be updated as soon as possible to ensure that all functions
- * that can be are turned into constexpr functions
- *
- * Example:
- * ENUMCLASS(Test, Value)
+ * if the enum class object is any good
+ * 
+ * Use check_enum_error(ENUMTYPE, message...) to throw an error
+ * composed by the variadic message if the enum value is bad
  *
- * Generates:
- * enum class Test : long {Value, FINAL};
- * namespace enumstrs {std::array<std::string, 1> TestNames={"Value"};};
- * std::string toString(Test x) noexcept;
- * Test toTest(const std::string& x) noexcept;
+ * \verbatim
+ // Example:
+ ENUMCLASS(Test, char, Value)
+ 
+ // Generates (effectively):
+ enum class Test : char {Value, FINAL};
+ namespace enumstrs {
+    constexpr std::array<std::string_view, 1> TestNames={"Value"};
+ }
+ namespace enumtyps {
+    constexpr std::array<Test, 1> TestTypes={Test::Value};
+ }
+ constexpr std::string_view toString(Test x) noexcept;
+ constexpr Test toTest(const std::string_view& x) noexcept;
+ inline std::ostream &operator<<(std::ostream &os, const Test x);
+ inline std::istream &operator>>(std::istream &is, Test &x);  // throws if not good_enum(x) at end
+ \endverbatim
  */
 #define ENUMCLASS(ENUMTYPE, TYPE, ...)                                    \
   enum class ENUMTYPE : TYPE { __VA_ARGS__, FINAL };                      \
                                                                           \
   namespace enumstrs {                                                    \
   constexpr auto ENUMTYPE##Names = enum_strarray<ENUMTYPE>(#__VA_ARGS__); \
+  }                                                                       \
+  namespace enumtyps {                                                    \
+  constexpr auto ENUMTYPE##Types = enum_typarray<ENUMTYPE>();             \
```

Comment:
This definition needs to be preceded by `[[maybe_unused]]` to silence these warnings:

```
/Users/olemke/Hacking/uhh/arts/src/linemixing.cc:120:1: warning: unused variable 'SpecialParamTypes' [-Wunused-const-variable]
ENUMCLASS(SpecialParam, char,
^
/Users/olemke/Hacking/uhh/arts/src/enums.h:155:18: note: expanded from macro 'ENUMCLASS'
  constexpr auto ENUMTYPE##Types = enum_typarray<ENUMTYPE>();             \
                 ^
<scratch space>:26:1: note: expanded from here
SpecialParamTypes
^
1 warning generated.
```

```
/Users/olemke/Hacking/uhh/arts/src/m_absorptionlines.cc:410:1: warning: unused variable 'HitranTypeTypes' [-Wunused-const-variable]
ENUMCLASS(HitranType, char,
^
/Users/olemke/Hacking/uhh/arts/src/enums.h:155:18: note: expanded from macro 'ENUMCLASS'
  constexpr auto ENUMTYPE##Types = enum_typarray<ENUMTYPE>();             \
                 ^
<scratch space>:23:1: note: expanded from here
HitranTypeTypes
^
1 warning generated.
```